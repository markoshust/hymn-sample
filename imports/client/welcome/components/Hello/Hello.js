import React, { PropTypes } from 'react';
import Button from 'client/core/components/Button';

const Hello = ({
  counter,
  onClick,
}) => (
  <div>
    <Button
      label="Click Me"
      onClick={onClick}
    />
    <p>You've pressed the button {counter} times.</p>
  </div>
);

Hello.propTypes = {
  counter: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Hello;
