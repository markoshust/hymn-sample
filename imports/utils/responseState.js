const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
