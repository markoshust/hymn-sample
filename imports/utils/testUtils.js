import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
