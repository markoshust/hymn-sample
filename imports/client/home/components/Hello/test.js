import React from 'react';
import { renderCheck } from '/imports/utils/testChecks';
import Hello from './Hello';

const onClick = jest.fn();

describe('<Hello />', () => {
  it('renders correctly', () => {
    renderCheck(
      <Hello
        counter={0}
        onClick={onClick}
      />
    );
  });
});
