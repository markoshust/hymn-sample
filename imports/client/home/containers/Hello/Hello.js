import { compose, mapProps, withState } from 'recompose';
import Hello from '../../components/Hello';

const enhance = compose(
  withState('counter', 'incCounter', 0),
  mapProps(({ incCounter, ...rest }) => ({
    onClick: () => incCounter(n => n + 1),
    ...rest,
  })),
);

export default enhance(Hello);
