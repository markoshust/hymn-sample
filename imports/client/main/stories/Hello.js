import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { number } from '@kadira/storybook-addon-knobs';
import Hello from '../components/Hello';

const stories = storiesOf('Hello', module);

stories.add('Hello', () => (
  <Hello
    counter={number('Counter', 0)}
    onClick={action('clicked')}
  />
));
