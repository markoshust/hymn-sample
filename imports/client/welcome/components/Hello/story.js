import React from 'react';
import { action, storiesOf } from '@kadira/storybook';
import { number } from '@kadira/storybook-addon-knobs';
import Hello from './Hello';

const stories = storiesOf('welcome/Hello', module);

stories.add('Hello', () => (
  <Hello
    counter={number('Counter', 0)}
    onClick={action('click')}
  />
));
