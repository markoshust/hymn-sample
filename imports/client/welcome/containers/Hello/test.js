import React from 'react';
import find from 'lodash/find';
import { componentCheck, renderCheck } from 'utils/testChecks';
import Hello from './Hello';

describe('Container(<Hello />)', () => {
  it('updates counter', () => {
    const { component, tree } = renderCheck(<Hello />);
    const Button = find(tree.children, { type: 'button' });

    Button.props.onClick();

    componentCheck(component);
  });
});
