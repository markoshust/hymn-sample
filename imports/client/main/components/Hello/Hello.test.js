import React from 'react';
import { renderCheck } from '/imports/utils/testUtils';
import Hello from './Hello';

const onClick = jest.fn();

describe('<Hello />', () => {
  it('renders correctly', () => {
    renderCheck(
      <Hello
        counter={0}
        onClick={onClick}
      />
    );
  });
});
