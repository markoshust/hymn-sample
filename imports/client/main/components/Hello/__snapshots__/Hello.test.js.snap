exports[`<Hello /> renders correctly 1`] = `
<div>
  <button
    onClick={[Function anonymous]}>
    Click Me
  </button>
  <p>
    You've pressed the button 
    0
     times.
  </p>
</div>
`;
